fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/InstrumentedReadTransaction.java

```diff
@@ -63,10 +65,24 @@
 
     protected final boolean enableAssertions;
 
+    private final long startNanos;
+
+    // Set of metrics that are accumulated and published once over the entire life of the transaction to allow for
+    // an operations-per-transaction view of things, even if the overall store timer is accumulating stats across
+    // multiple transactions.
+    private boolean transactionMetricsPublished = false;
+    protected final AtomicInteger transactionReads = new AtomicInteger();
+    protected final AtomicInteger transactionWrites = new AtomicInteger();
+    protected final AtomicInteger transactionDeletes = new AtomicInteger();
+    protected final AtomicInteger transactionBytesRead = new AtomicInteger();
+    protected final AtomicInteger transactionBytesWritten = new AtomicInteger();
+    protected final AtomicInteger transactionMutations = new AtomicInteger();
```

Comment:
Do you want per-transaction `EventKeeper` info, too? If so, I think the basic idea would be to pull this out into a separate class, move the interface implementation onto that and init the various objects in the right order in `FDBDatabase.createTransaction`.

Reply:
I completely re-did the way it works.  Not only are you correct that the EventKeeper data is useful (so I changed how it works), but that my goal of capturing I/O distributions by transaction wasn't reasonably feasible the way I had things implemented.  The new commit has a description of my thinking in it.